src/webServer/webServer.ts

```diff
@@ -136,7 +138,34 @@ namespace ts.server {
             clearImmediate: handle => clearTimeout(handle),
             /* eslint-enable no-restricted-globals */
 
-            require: () => ({ module: undefined, error: new Error("Not implemented") }),
+            require: (initialPath: string, moduleName: string) => {

```

Comment:
Looking over the TS code, I think `require` is only used for testing and [loading plugins](https://github.com/microsoft/TypeScript/blob/009dd487a6543d16c8e403ea055ac0683c6bc5d3/src/server/project.ts#L232)

If that's the case, I believe we should  introduce a more scoped implementation instead of `require`, something like: `loadPluginModule` instead? On web, I think we really want to be able to implement [`resolveModule`](https://github.com/microsoft/TypeScript/blob/009dd487a6543d16c8e403ea055ac0683c6bc5d3/src/server/project.ts#L232) ourselves